sa/model.go

```diff
@@ -1055,26 +1055,29 @@ func deleteOrderFQDNSet(
 }
 
 func addIssuedNames(ctx context.Context, queryer db.Execer, cert *x509.Certificate, isRenewal bool) error {
-	if len(cert.DNSNames) == 0 {
-		return berrors.InternalServerError("certificate has no DNSNames")
-	}
-
-	multiInserter, err := db.NewMultiInserter("issuedNames", []string{"reversedName", "serial", "notBefore", "renewal"})
-	if err != nil {
-		return err
-	}
-	for _, name := range cert.DNSNames {
-		err = multiInserter.Add([]interface{}{
-			ReverseName(name),
-			core.SerialToString(cert.SerialNumber),
-			cert.NotBefore.Truncate(24 * time.Hour),
-			isRenewal,
-		})
+	// TODO(#7311): Determine & explicitly document whether to place IP address
+	// identifiers in issuedNames. We currently skip them.
+	if len(cert.DNSNames) > 0 {
+		multiInserter, err := db.NewMultiInserter("issuedNames", []string{"reversedName", "serial", "notBefore", "renewal"})
 		if err != nil {
 			return err
 		}
+		for _, name := range cert.DNSNames {
+			err = multiInserter.Add([]interface{}{
+				ReverseName(name),
+				core.SerialToString(cert.SerialNumber),
+				cert.NotBefore.Truncate(24 * time.Hour),
+				isRenewal,
+			})
+			if err != nil {
+				return err
+			}
+		}
+		return multiInserter.Insert(ctx, queryer)
+	} else if len(cert.IPAddresses) > 0 {
+		return nil
 	}
-	return multiInserter.Insert(ctx, queryer)
+	return berrors.InternalServerError("certificate has no DNSNames or IPAddresses")
```

Comment:
I'm generally not a fan of "if we got to the end of the function, that's an error". I think it's easy to have a `if len(cert.DNSNames) == 0 && len(cert.IPAddresses) == 0 { return error }` at the top.